clang/include/clang/Sema/Overload.h

```diff
@@ -1015,6 +1045,62 @@ class Sema;
           RewriteKind(CRK_None) {}
   };
 
+  struct DeferredTemplateOverloadCandidate {
+    DeferredTemplateOverloadCandidate *Next = nullptr;
```

Comment:
Comment here + on the slab allocator function as to WHY we are doing this instead of a std::vector/etc would probably be helpful.  I understand it, but I probably won't next time I see this code ;)